### PR TITLE
Added universal2 wheels

### LINKED ISSUE #359: universal2 wheels for macOS
Hello! It would be really helpful to have `universal2` builds of Pillow available for macOS in addition to the `x86_64` and `arm64` builds. 

Having a single wheel that supports both platforms makes it a lot easier to release tools across both macOS architectures.

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #359 by adding universal2 wheels, apart from PyPy.

We don't build PyPy versions for arm64, so they're not going to work for universal2.

### CODE REVIEW

File: config.sh [hunk 1]
```diff
@@ -27,6 +27,35 @@ BZIP2_VERSION=1.0.8
...
+       if [[ "$whl" == *macosx_${py_osx_ver}_x86_64.whl ]]; then
+           whl_base=$(sed "s/macosx_${py_osx_ver}_x86_64.whl//" <<< $whl)
+           if [[ -f "${whl_base}macosx_11_0_arm64.whl" ]]; then
+               delocate-fuse $whl "${whl_base}macosx_11_0_arm64.whl" -w tmp_fused_wheelhouse
+               mv tmp_fused_wheelhouse/$(basename $whl) $wheelhouse/$(basename ${whl_base})macosx_${py_osx_ver}_universal2.whl
+               rm $whl
+           fi
+       fi
+    done
+}
```
Comment (radarhere) [inline config.sh:44]: This change has been approved, but merged, in https://github.com/multi-build/multibuild/pull/495

File: config.sh [hunk 2]
```diff
@@ -27,6 +27,35 @@ BZIP2_VERSION=1.0.8
...
+        (macos_arm64_cross_build_setup && $@)
+        rm -rf *-stamp
+        (macos_intel_native_build_setup && $@)
+        fuse_macos_intel_arm64
+    elif [[ "${PLAT:-}" == "arm64" ]]; then
+        (macos_arm64_cross_build_setup && $@)
+    elif [[ "${PLAT:-}" == "x86_64" ]]; then
+        (macos_intel_native_build_setup && $@)
+    fi
+}
```
Comment (radarhere) [inline config.sh:57]: There is debate about this in https://github.com/multi-build/multibuild/pull/495

File: config.sh [hunk 3]
```diff
@@ -52,8 +81,10 @@ function build_brotli {
 function pre_build {
     # Any stuff that you need to do before you start building the wheels
     # Runs in the root directory of this repository.
-    curl -fsSL -o pillow-depends-main.zip https://github.com/python-pillow/pillow-depends/archive/main.zip
-    untar pillow-depends-main.zip
+    if [ ! -d pillow-depends-main ]; then
+        curl -fsSL -o pillow-depends-main.zip https://github.com/python-pillow/pillow-depends/archive/main.zip
+        untar pillow-depends-main.zip
+    fi
```
Comment (radarhere) [inline config.sh:87]: The build is run twice for universal2, once for x86_64 and once for arm64, and pillow-depends only needs to be downloaded once.

File: .github/workflows/build.sh [hunk 4]
```diff
@@ -32,6 +32,9 @@ echo "::endgroup::"
 
 echo "::group::Build wheel"
   clean_code
+  if [[ "$PLAT" == "universal2" ]]; then
+    export MB_PYTHON_OSX_VER="10.10"
```
Comment (nulano) [inline .github/workflows/build.sh:36]: Can this be moved to be next to the other `MB_PYTHON_OSX_VER` export?

https://github.com/python-pillow/pillow-wheels/blob/c5def31b0f72ad620c6a28fa5e9daa4833d861f1/.github/workflows/build.sh#L17-L18

Comment (radarhere) [reply .github/workflows/build.sh:36]: It's not quite as simple as that - https://github.com/radarhere/pillow-wheels/commit/82dba2b0d64f2667f7ca59d712f5016ef9ac8dae / https://github.com/radarhere/pillow-wheels/actions/runs/4024580904/jobs/6916784545

File: config.sh (hunk 1 above)
Comment (radarhere) [reply config.sh]: https://github.com/multi-build/multibuild/pull/495 has been merged now, so I've updated multibuild to include it, and removed this patch.

File: .github/workflows/build.sh (hunk 4 above)
Comment (radarhere) [reply .github/workflows/build.sh]: That PR has been merged, so I've removed this code.

File: config.sh (hunk 2 above)
Comment (radarhere) [reply config.sh]: I've pushed a commit as an alternative to this, making use of https://github.com/multi-build/multibuild/pull/502

**Comment by radarhere** [comment]
Closing in light of https://github.com/python-pillow/pillow-wheels/issues/359#issuecomment-1600382015

**Pull request closed without merging**